kroxylicious-api/src/main/java/io/kroxylicious/proxy/filter/FilterThreadExecutor.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.filter;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ScheduledExecutorService;
+
+/**
+ * Kroxylicious guarantees that each Filter instance has a single thread associated with it (the same
```

Comment:
We want the package doc to the source of truth for the core principles (such as how threading works)  The classdoc can then make references to the packagedoc.  We'll minimise inconsistency that way.